libcanard/canard.c

```diff
@@ -340,8 +340,8 @@ CANARD_PRIVATE struct CanardTxQueueItem* txAllocateQueueItem(struct CanardTxQueu
 /// Frames with identical CAN ID that are added later always compare greater than their counterparts with same CAN ID.
 /// This ensures that CAN frames with the same CAN ID are transmitted in the FIFO order.
 /// Frames that should be transmitted earlier compare smaller (i.e., put on the left side of the tree).
-CANARD_PRIVATE int8_t txAVLPriorityPredicate(           //
-    void* const                        user_reference,  // NOSONAR Cavl API requires pointer to non-const.
+CANARD_PRIVATE ptrdiff_t txAVLPriorityPredicate(        //
+    const void*                        user_reference,  // NOSONAR Cavl API requires pointer to non-const.
```

Comment:
```suggestion
    const void* const                  user_reference,
```

It doesn't require that anymore, does it? Please update the other occurrences as well. Also the pointer value itself should be `const`, too.